### PR TITLE
feat(dot): CW-19129 update dot bond function

### PR DESCRIPTION
**Pull request by hankliu418**
----

*Checklist:*

- README.md includes:
  - [ ] The details of signing data and transaction data, this includes parameters.
  - [ ] Official docs or white paper.
  - [ ] Website or api can query assets and broadcast transaction.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by cbx-tech** [comment]
Task linked: [CW-19129 [SDK] 因 Polkadot 升級使 bond 送交易失敗](https://app.clickup.com/t/25577573/CW-19129)

File: packages/coin-dot/src/utils/dotUtil.ts [hunk 1]
```diff
@@ -37,123 +35,136 @@ export function getFormatTxData(rawData: types.dotTransaction): types.FormatTran
...
     method: {
       callIndex,
-      controllerAddress,
       value,
-      payeeType
+      payeeType,
     },
-    methodString: callIndex + params.TX_ADDRESS_PRE + controllerAddress + formatSCALECodec(rawData.value) + payeeType
-  }
+    methodString: callIndex + formatSCALECodec(rawData.value) + payeeType,
```
Comment (tp26610) [inline packages/coin-dot/src/utils/dotUtil.ts:74]: 只有改這邊，其他是 format

**Review by tp26610 — APPROVED** [review]

**Review by gundomlegend — APPROVED** [review]

File: packages/coin-dot/src/config/params.ts [hunk 2]
```diff
@@ -47,99 +51,97 @@ export const SCRIPT_PARAMS = {
...
+        '00003044022037AC2FFEEA6AF1DC07793EDBAC2FF1B6DE1E2812AD78D4C0D3CB950C6913726F022064370FC4A38247BB531396AD3C866DDB5A5350150C542806D9C1994A76F86645',
     },
     BOND: {
       script:
-        '03020E01C7070000000162CAA0C70002CC071000CAAC570002A2ACD70022FFF6CAAC17002CCAACD7002DFFFBA2ACD70032FFFBA2ACD70037FFFBCAACC7003C04CAA6C70004CAAC570044CAAC570064DC07C003444f54DC07C004426f6e64CC0FC00753533538505245CC0F1000CAAC5F00025AF09FC00FBAFCCE6C07230804DDE09700DAACD7C022FFF60AD207CC05065052455353425554546f4e',
+        '03020E01C7070000000162CAA0C70002A2ACD70002FFF6CAAC17000CCAACD7000DFFFBA2ACD70012FFFBA2ACD70017FFFBCAACC7001C04CAA5C70004CAAC570024CAAC570044DC07C003444F54DC07C004426F6E64DAACD7C002FFF60AD207CC05065052455353425554546F4E',
       signature:
-        '00304502202199859BC3BB3C2ACC00272286B8D3FA3E326DC807553D15669FC22D7B82B8F8022100BE967FEA54A9466850C360BE24EFE481975234B650F9ED5EAEBFD06E80AC9BA4'
+        '3046022100cc5125c9f6ea1ff6c455e7903cbee0842689632fa30e3a2e145017e6c818f5580221008d614c6f29444f074ad9091b2334aaf3690e9fecc11d28e53603c16c9975970f',
```
Comment (tp26610) [inline packages/coin-dot/src/config/params.ts:60]: script 調整，拿掉 controller address

File: packages/coin-dot/src/config/types.ts [hunk 3]
```diff
@@ -62,7 +62,6 @@ export interface BondTx extends dotTransaction {
   method: BondMethod;
 }
 export interface BondMethod {
-  controllerAddress: string;
```
Comment (tp26610) [inline packages/coin-dot/src/config/types.ts:65]: 拿掉 controllerAddress 定義

File: packages/coin-dot/src/utils/dotUtil.ts [hunk 4]
```diff
@@ -174,40 +185,37 @@ export function getMortalEra(blockNumber: string, era: string): string {
...
       break;
     case params.ValueMode.foreByteMode:
-      formatValue = (bigValue.shln(2)).or(new BN('2'))
+      formatValue = bigValue.shln(2).or(new BN('2'));
       break;
     case params.ValueMode.bigIntegerMode:
-      const length = Math.ceil(bigValue.toString(16).length / 2)
-      const addCode = (length - 4).toString(2).padStart(6, '0') + '11'
+      const length = Math.ceil(bigValue.toString(16).length / 2);
+      const addCode = (length - 4).toString(2).padEnd(6, '0') + '11';
```
Comment (tp26610) [inline packages/coin-dot/src/utils/dotUtil.ts:204]: 這邊改錯，下個 PR 修正

**Pull request merged**